packages/relay-compiler/core/RelayIR.js

```diff
@@ -158,7 +157,7 @@ export type ScalarField = {
   kind: 'ScalarField',
   metadata: ?{[key: string]: mixed},
   name: string,
-  type: GraphQLLeafType,
+  type: GraphQLOutputType,
```

Comment:
This seems wrong: one of the major reasons we added LinkedField/ScalarField in the first place was to distinguish between values whose types were leaf-like and those that have subfields. Is there any reason not to be more explicit here?

```
type ScalarType = GraphQLLeafType | GraphQLNonNull<ScalarType> | GraphQLList<ScalarType>;

type ScalarField = {
   ...
  type: ScalarType,
}
```